app/models/schedule.rb

```diff
@@ -137,4 +137,18 @@ def clean_up_task_jobs
     end
   end
 
+  def timeline
+    timeline = []
+    self.root_task.self_and_descendants.each do |task|
+      timeline.push(["#{task.equipment.type.titlecase}", "#{task.type}"[0..-5], task.trigger_start*1000/60, (task.trigger_start + task.duration)*1000/60])
```

Comment:
You need a check to see if task.equipment.nil? is true. I think in most cases that won't happen due to validation on Schedule, but I get an error on the Recipes Index page because somehow I have two Tasks without no Equipment set.